core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2008,8 +2031,7 @@ public MergeStats getMergeStats() {
         return mergeScheduler.stats();
     }
 
-    @Override
-    public SequenceNumbersService seqNoService() {
+    public final SequenceNumbersService seqNoService() {
```

Comment:
Just wondering: Is it possible to move `seqNoService` field to `Engine` make the `seqNoService()` method there final? The `SeqNoServiceSupplier` (or BiFunction) should then also be moved as an constructor parameter to `Engine`.

Reply:
I don't think we can do this cleanly right now. It means that the sequence number service has to be available immediately (to pass to the super constructor) but we are also trying to initialize the index writer at the same time that we initialize the sequence number service.

Reply:
I see, that makes it trickier. Thanks for explaining that.